codegen/smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/model/SymbolExt.kt

```diff
@@ -92,27 +100,31 @@ val Symbol.propertyTypeMutability: PropertyTypeMutability?
 
 /**
  * Gets the default value for the symbol if present, else null
- * @param defaultBoxed the string to pass back for boxed values
+ * @param defaultNullable the string to pass back for nullable values
  */
-fun Symbol.defaultValue(defaultBoxed: String? = "null"): String? {
-    // boxed types should always be defaulted to null
-    if (isBoxed) {
-        return defaultBoxed
-    }
-
+fun Symbol.defaultValue(defaultNullable: String? = "null"): String? {
     val default = getProperty(SymbolProperty.DEFAULT_VALUE_KEY, String::class.java)
-    return if (default.isPresent) default.get() else null
+    val defaultType = default.getOrNull()?.run { getProperty(SymbolProperty.DEFAULT_VALUE_TYPE_KEY, DefaultValueType::class.java).get() }
```

Comment:
**Style**: I suggest a convenience extension property for getting the default type, similar to `isNullable`.